packages/js-sdk/src/Dm3Sdk.test.ts

```diff
@@ -1,34 +1,278 @@
+import { StorageAPI } from '@dm3-org/dm3-lib-storage';
+import {
+    getMockDeliveryServiceProfile,
+    MockDeliveryServiceProfile,
+    MockedUserProfile,
+    MockMessageFactory,
+    mockUserProfile,
+} from '@dm3-org/dm3-lib-test-helper';
+import axios from 'axios';
 import { ethers } from 'ethers';
 import { Dm3Sdk, Dm3SdkConfig } from './Dm3Sdk';
-import { StorageAPI } from '@dm3-org/dm3-lib-storage';
+
+import MockAdapter from 'axios-mock-adapter';
+import { normalizeEnsName } from '@dm3-org/dm3-lib-profile';
+import { ITLDResolver } from './tld/nameService/ITLDResolver';
 
 describe('Dm3Sdk', () => {
-    let upController: ethers.Signer;
+    let alice: MockedUserProfile;
+    let bob: MockedUserProfile;
+    let karl: MockedUserProfile;
+
+    //Axios mock to mock the http requests
+    let axiosMock;
+
+    let deliveryService: MockDeliveryServiceProfile;
 
     beforeEach(async () => {
-        upController = ethers.Wallet.createRandom();
+        alice = await mockUserProfile(
+            ethers.Wallet.createRandom(),
+            'alice.up',
+            ['test.io'],
+        );
+        bob = await mockUserProfile(ethers.Wallet.createRandom(), 'bob.up', [
+            'test.io',
+        ]);
+
+        karl = await mockUserProfile(ethers.Wallet.createRandom(), 'karl.up', [
+            'test.io',
+        ]);
+
+        deliveryService = await getMockDeliveryServiceProfile(
+            ethers.Wallet.createRandom(),
+            'http://localhost:3000',
+        );
+
+        axiosMock = new MockAdapter(axios);
+
+        //Mock BackendConnector HttpRequests
+        //Mock profileExistsOnDeliveryService
+        axiosMock
+            .onGet(
+                `http://localhost:4060/profile/${normalizeEnsName(
+                    alice.address,
+                )}.addr.test`,
+            )
+            .reply(200);
+        //Mock getChallenge
+        axiosMock
+            .onGet(
+                `http://localhost:4060/auth/${normalizeEnsName(
+                    alice.address,
+                )}.addr.test`,
+            )
+            .reply(200, 'mock-challenge');
+
+        //Mock getToken
+        axiosMock
+            .onPost(
+                `http://localhost:4060/auth/${normalizeEnsName(
+                    alice.address,
+                )}.addr.test`,
+            )
+            .reply(200);
+    });
+
+    describe('conversations', () => {
+        it('can add a conversation to the contact list', async () => {
```

Comment:
just for future tests, consider using `it should ...` pattern so that tests form a sentence in combination with `describe` e.g. "Testing conversations -> should add a conversation to the contact list"
And then the expect checks exactly for the "should" part.

Reply:
That is a good take i normally follow. When I continue with the API implementation I'll use a cleaner API naming convention.